sync/git.go

```diff
@@ -48,7 +48,7 @@ func (p GitTagSyncProvider) GetRevision(ctx context.Context) (string, error) {
 	}
 
 	if p.verifyTag {
-		if _, err := p.repo.VerifyTag(ctx, p.syncTag); err != nil {
+		if _, err := p.repo.VerifyTag(ctx, p.syncTag); err != nil && !strings.Contains(err.Error(), "not found.") {
```

Comment:
This is a change in behaviour compared to what it was _before_ this PR.

The previous check was to distinguish a verification error from a not found error, and in case of the later, it would verify all commits before continuing with the sync.

After this change, it returns the revision we failed to verify, and the sync will continue from this revision, opening up an attack vector.

Reply:
```suggestion
		if _, err := p.repo.VerifyTag(ctx, p.syncTag); err != nil {
		    strings.Contains(err.Error(), "not found.") {
		        return "", nil
		    }
		    return "", err
```

should do the job, I think.

Reply:
Well argued -- so, in the "not found" case, it should return an empty revision and `nil` error, right?

Reply:
Correct.